### PR TITLE
Fix renaming a variable at the start of a line

### LINKED ISSUE #63: Renaming merges two lines
## Info

NVIM v0.5.0-dev+1073-gb894fbcbe
jedi-language-server, version 0.21.0
Python 3.9.1

## Problem

Renaming a variable merges the line where the variable is defined with the line above.

## Minimal init.vim

```vim
set runtimepath^=~/.local/share/nvim/plugins/nvim-lspconfig
lua require 'lspconfig'.jedi_language_server.setup{}
nnoremap gr :lua vim.lsp.buf.rename()<cr>
```

## Minimal bug.py

```py
# bug incoming
x = 1
```

## Reproduce

1. Install neovim-nightly
1. Install [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig) and put it on nvims runtimepath. In my case its `~/.local/share/nvim/plugins/nvim-lspconfig`
1. Create minimal vimrc and bug.py
1. Open bug.py with neovim
1. Move cursor on the variable `x`
1. Type `gr` and rename variable to y

The comment and the variable are now merged into a single line:

```py
# bug incomingx = 1
```

I tried the same setup with the python-language-server and it worked without the bug.

### PR DESCRIPTION
**Pull request by dimbleby**
Since I was on a roll, I wondered about #63.

Seems like a straightforward out-by-one error where the first character of a line is wrongly considered to be on the previous line (and also on its own line; but we find the previous line first).

However, it's clear that the code that I am fixing was deliberately written that way - at b6abf85676bb1e5eae9ad79f2b050e8dfdd33e39.  Perhaps you'll remember the reason for that, and therefore what this breaks!

Fixes #63.

### CODE REVIEW

**Comment by dimbleby** [comment]
(there will be minor merge conflicts with #78: in the CHANGELOG, and because of the slight refactoring of the initialization of LspSession.  Should be trivial, but happy to handle if and when you merge one or the other of these first)

File: tests/lsp_tests/test_rename.py [hunk 1]
```diff
@@ -78,3 +78,38 @@ def test_lsp_rename_function():
             ],
         }
         assert_that(actual, is_(expected))
+
+
+def test_lsp_rename_variable_at_line_start():
+    """Tests renaming a variable that appears at the start of a line."""
+    with session.LspSession() as ls_session:
+        uri = as_uri((REFACTOR_TEST_ROOT / "rename_test2.py"))
```
Comment (pappasam) [inline tests/lsp_tests/test_rename.py:86]:
```suggestion
    with session.LspSession() as ls_session:
        ls_session.initialize()
        uri = as_uri((REFACTOR_TEST_ROOT / "rename_test2.py"))
```

Comment (pappasam) [reply tests/lsp_tests/test_rename.py:86]: Tests don't run unless you initialize the ls session here

**Review by pappasam — APPROVED** [review]
I don't remember the exact reason why I added that +1. I think I was shifting things around and manually testing with `coc` at the time; once things worked, I went to bed and called it a day. So, based on your test case (I'll add the `ls_session` piece after merging) and my own manual testing, this doesn't seem to add any regressions and it looks like it fixes the issue you've referenced. Thanks a lot for fixing this, since coc seems to weirdly handle this, I would have had a hard time getting to the bottom of this.

**Pull request merged**